### DISCUSSION TITLE
如何设置自定义解析（如r.example.com解析到192.168.100.1）

### DISCUSSION THREAD

**Discussion by lr7skquo** [kkkgo/PaoPaoDNS#27] [问答和帮助] [answered]
如题。
看说明文档，可以通过“USE_HOSTS配合docker的-add-hosts或者docker compose的extra_hosts使用”达到此目的。但在qnap nas下通过ui配置（且不使用docker compose）无法实现。
请问能否通过修改/data目录下某个设置文件实现此功能？

**Comment by kkkgo** [comment]
docker的-add-hosts或者docker compose的extra_hosts功能原理都是在docker启动的时候把内容写入docker里面的hosts文件，然后USE_HOSTS功能开启后会读取容器的hosts文件。如果你可以挂载hosts文件出来的话倒也可以修改，不过威联通的UI似乎没有提供挂载文件的功能，只有挂载文件夹。如果你要使用威联通的docker compose功能，可以参考仓库里面的威联通docker compose配置文件https://github.com/kkkgo/PaoPaoDNS/blob/main/docker-compose-qnap.yaml
或者安装portainer等第三方docker管理ui。   
此外，下一个版本将会在不久后发布，会有更多自定义功能，可以实现修改/data目录下某个设置文件实现此功能。🙂

**Comment by kkkgo** [comment, marked as answer]
新版本已发布，你可以使用`RULES_TTL`功能实现你的功能。可以查阅更新日志：https://github.com/kkkgo/PaoPaoDNS/discussions/28
